### ISSUE TITLE
year is twice there

### ISSUE THREAD

**Issue by mejem** [michal-h21/biblatex-iso690#15]
[EN]
Hi,
according to the ISO 690 standard, the bibliography entry shouldn't look like this:

CHOMSKY, Noam. 1959. A review of B. F. Skinner’s Verbal Behavior. Language. 1959, roč. 35, č. 1, s. 26–58.

There shouldn't be year after author. The year is twice there.

This happens only when you use "style=iso-authoryear" in biblatex.
When you use numeric, it works fine.
Reference in text for authoryear should look like this: Chomsky, Noam, 1959

Could you fix this, please?
Thank you.

[CZ]
Zdravím,
chtěl bych poděkovat za super kódy.
Chtěl bych to použít u bakalářky, ale je tam jedna drobnost:
SKINNER, B. F. 1957. Verbal behavior. 1957. Century psychology series.
Ten rok za jménem být podle normy nemá.
Pozor ale, aby nezmizel rok i z odkazů v textu.
Šlo by to nějak opravit?
Děkuju.

**Comment by johannesbottcher** [comment]
Hi, i am also watching here and try to help out, unfortunately i don't speak czech. Can you rephrase in english? I might be able to help you.

**Comment by mejem** [comment]
Hi,
according to the ISO 690 standard, the bibliography entry shouldn't look like this:

CHOMSKY, Noam. 1959. A review of B. F. Skinner’s Verbal Behavior. Language. 1959, roč. 35, č. 1, s. 26–58.

There shouldn't be year after author. The year is twice there.
Could you fix this, please?
Thank you.

**Comment by mejem** [comment]
Hi,
according to the ISO 690 standard, the bibliography entry shouldn't look like this:

CHOMSKY, Noam. 1959. A review of B. F. Skinner’s Verbal Behavior. Language. 1959, roč. 35, č. 1, s. 26–58.

There shouldn't be year after author. The year is twice there.
Could you fix this, please?
Thank you.

**Comment by mejem** [comment]
This happens only when you use "style=iso-authoryear" in biblatex.
When you use numeric, it works fine.
Reference in text for authoryear shoul look like this: Chomsky, Noam, 1959

**Comment by michal-h21** [comment]
It used to be there twice in the older ISO version (or at least in the
Czech interpretation), it is one of a few good and logical changes in the
new version. In the authoryear style, reference should look like this, I
think (I can find examples only for books, but year should be after the
author and the second appearance should be removed):

CHOMSKY, Noam, 1959. A review of B. F. Skinner’s Verbal Behavior. Language.
Roč. 35, č. 1, s. 26–58.

I've created a fix for this and uploaded changes to Github.

Best,
Michal

**Comment by mejem** [comment]
OK then, thank's.

And can you please tell me, how can I make references within text look like this:
Chomsky, 1959
without first name?

**Comment by michal-h21** [comment]
I've tried something, it print only surname now, but only in uppercase. I
can't find a easy way to fix that without removing upper case names in the
bibliography. Maybe Johaness know?

**Comment by mejem** [comment]
I have solved this:

\ProvidesFile{iso-authoryear.cbx}[2011/06/08]
\ExecuteBibliographyOptions{%
	 maxcitenames=1
	,mincitenames=1
}

\RequireCitationStyle{authoryear}
\DeclareNameFormat{labelname}{#1}

Now it works as I want.
Thank you for your time and effort.

**Comment by johannesbottcher** [comment]
@mejem  Try an entry with multiple authors. ;-)

@michal-h21  Strange styles that define the appearance of a cite in the `bbx` ;-) Makes debugging a pain.

**Comment by mejem** [comment]
In combination with
natbib=true
(I want to use also \citep and \citet)
it works fine. :)

**Comment by michal-h21** [comment]
> @mejem <https://github.com/mejem> Try an entry with multiple authors. ;-)
>

yeah, this problem is still there :(

> @michal-h21 <https://github.com/michal-h21> Strange styles that define
> the appearance of a cite in the bbx ;-) Makes debugging a pain.
>

we don't define appearance of cite in the bbx, only appearance of surrnane,
but this is also used in citations :(

**Comment by johannesbottcher** [comment]
@michal-h21  Looking into this again i noticed, that the styles require the generic `iso.bbx`, but it cannot work by itself. Commands like `MethodFormat` are commented out and introduced (defined) within the more special styles. I am again thinking, that resetting the whole thing now with version 3 of biblatex might be the most robust version. 

Though i never really understood that issue with the `online` entries (see https://github.com/michal-h21/biblatex-iso690/issues/12#issuecomment-108626275).

**Comment by johannesbottcher** [comment]
@mejem  This would be what you suggested as a fix, what do you get running this example? 

```latex
\documentclass{article}
\usepackage[czech]{babel}
\usepackage[
style=iso-authoryear,
natbib=true
]{biblatex}
\addbibresource{biblatex-examples.bib}
\DeclareNameFormat{labelname}{#1}
\begin{document}
\cite{companion}
\printbibliography
\end{document}
```

**Comment by mejem** [comment]
https://dl.dropboxusercontent.com/u/84775532/one.pdf

**Comment by johannesbottcher** [comment]
@mejem  Right, `maxcitenames=1`. That is also related to https://github.com/michal-h21/biblatex-iso690/issues/12#issue-73364898

I wish i had more time to really tackle that whole thing completely instead of single instances. :-(
